### ISSUE TITLE
Golden Sun Crashing (SegFault) using mgba libretro on a raspberry pi

### ISSUE THREAD

**Issue by MJLHThomassen** [mgba-emu/mgba#464]
Hello. I've been having some issues with Golden Sun, I don't feel like typing the whole story for the 3rd time so more background info here:

https://retropie.org.uk/forum/topic/4967/golden-sun-gba-crashing/7

and here:

https://github.com/libretro/mgba/issues/47

Short version: Golden Sun.gba (ROM MD5: 1c dd 7f 33 c9 a2 70 61 20 1e 1d 1c e0 29 a7 00) crashes at seeminly random points/locations/moments in the game. I've been able to get a backtrace of a coredump after some trying which looks like the following:

```
#0  0x75e2f224 in GBAVideoSoftwareRendererPreprocessSprite () from /opt/retropie/libretrocores/lr-mgba/mgba_libretro.so
No symbol table info available.
#1  0x75e35074 in GBAVideoSoftwareRendererDrawScanline () from /opt/retropie/libretrocores/lr-mgba/mgba_libretro.so
No symbol table info available.
#2  0x75e21298 in GBAVideoProcessEvents () from /opt/retropie/libretrocores/lr-mgba/mgba_libretro.so
No symbol table info available.
#3  0x75e1ec9c in GBAProcessEvents () from /opt/retropie/libretrocores/lr-mgba/mgba_libretro.so
No symbol table info available.
#4  0x75e1d430 in _GBACoreRunFrame () from /opt/retropie/libretrocores/lr-mgba/mgba_libretro.so
No symbol table info available.
#5  0x75e3d904 in retro_run () from /opt/retropie/libretrocores/lr-mgba/mgba_libretro.so
No symbol table info available.
#6  0x0001f50c in core_run () at core_impl.c:411
No locals.
#7  0x00033f58 in runloop_iterate (sleep_ms=0x7ef0e5d4, sleep_ms@entry=0x7ef0e5cc) at runloop.c:1233
        i = <optimized out>
        current = <optimized out>
        target = <optimized out>
        to_sleep_ms = <optimized out>
        last_input = 64
        runloop_status = <optimized out>
        frame_limit_minimum_time = 16743
        frame_limit_last_time = 3751011610
        settings = 0x76667008
        current_input = 64
        old_input = 0
#8  0x0001e6c8 in rarch_main (argc=<optimized out>, argv=<optimized out>, data=0x0) at frontend/frontend.c:127
        sleep_ms = 0
        ret = <optimized out>
        argc = <optimized out>
        argv = <optimized out>
        data = 0x0
        args = 0x0
#9  0x76a76294 in __libc_start_main (main=0x7ef0e754, argc=1991880704, argv=0x76a76294 <__libc_start_main+276>, init=<optimized out>, 
    fini=0x129108 <__libc_csu_fini>, rtld_fini=0x76fd8408 <_dl_fini>, stack_end=0x7ef0e754) at libc-start.c:287
        self = <optimized out>
        result = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {0, 0, 114592, 0, 0, 0, 1996455936, 0, -1908578065, -2039812929, 0 <repeats 16 times>, 1993490432, 102960, 
                0, 1996458768, 1993494792, 1993753264, -1, 102828, 1990630952, 1992136080, 1993494792, 1, 0, 1996321720, 0, 1, 1, 0, 2129717104, 1993493224, 
                0, 1996458328, 0, 1806912, 1987630272, 0, 76580, 1, 0, 1996321720, 1987762392, 1, 1, 0, 82928, 1990630952, 0, 0}, mask_was_saved = 0}}, 
          priv = {pad = {0x0, 0x0, 0x76ff8000, 0x76fddf50 <_dl_runtime_resolve+24>}, data = {prev = 0x0, cleanup = 0x0, canceltype = 1996455936}}}
        not_first_call = <optimized out>
#10 0x0001bfcc in _start ()
No symbol table info available.
```

Then after this the emulator gives a SegFault.

**Comment by endrift** [comment]
I don't seem to be able to reproduce the issue, in debugger, out of debugger, with libretro core, without libretro core. Anything. It just doesn't crash here. I'll try a few more things later, but...this is a weird one.

**Comment by endrift** [comment]
This was fixed by 3415484d9db5528bfbeb71a5bf0bb30d0eeef410
